### PR TITLE
Pull Request for Issue2402: Fixes to the Ge detector view visibility toggling.

### PR DESCRIPTION
**Pull request by helfrij**

### CODE REVIEW

**Comment by helfrij** [comment]
Hey @XuetaoLiu-CLS, would you review?

File: source/application/BioXAS/BioXASAppController.cpp [hunk 1]
```diff
@@ -635,6 +631,16 @@ AMScanConfigurationViewHolder3* BioXASAppController::createScanConfigurationView
 	return configurationViewHolder;
 }
 
+void BioXASAppController::addComponentView(QObject *component, const QString &viewName, const QString &categoryName, const QString &icon)
```
Comment (XuetaoLiu-CLS) [inline source/application/BioXAS/BioXASAppController.cpp]: Is it possible to move this function to the `CLSAppController`? Then we can have the `componentViewMapping_` in CLSAppController, add add a function to do the following mapping:
```
    QWidget *detectorView = componentViewMapping_.value(detector, 0);
    QWidget *detectorPane = viewPaneMapping_.value(detectorView, 0);
```

Maybe the other beamlines can use it as well.

**Comment by XuetaoLiu-CLS** [comment]
Looks good, other than a question on code generalization.

Time Estimate: 5 minutes

File: source/application/BioXAS/BioXASAppController.cpp (hunk 1 above)
Comment (helfrij) [inline source/application/BioXAS/BioXASAppController.cpp]: It's possible, we can take a couple of moments to investigate together if you like.

**Comment by dretrex** [comment]
Looks good to me.
Time Estimate: 1 minute

**Pull request merged**